vivisect/analysis/ppc/bootstrap.py

```diff
@@ -1,41 +1,46 @@
-# this analysis module is for blobs and ihex files, and attempts to identify boot information as understood by a Boot Assist Module (BAM) found in many of the modern PPC/VLE processors.
-# for reference, this module uses information from the "MPC5674F Microcontroller Reference Manual, rev 7" from NXP, as well as "MPC5646C Microcontroller Reference Manual, rev. 5".
-# we won't mimic the behavior exactly, but rather use BAM information to identify entry points
+# This analysis module is for blobs and ihex files, and attempts to identify
+# boot information as understood by a Boot Assist Module (BAM) found in many of
+# the modern PPC/VLE processors.
+#
+# For reference, this module uses information from the following reference
+# manuals:
+#   - "MPC5674F Microcontroller Reference Manual, rev 7" from NXP
+#   - "MPC5646C Microcontroller Reference Manual, rev. 5" from NXP
+#   - "MPC5777C Microcontroller Reference Manual, rev. 8" from NXP
+#
+# We won't mimic the behavior exactly, but rather use BAM information to
+# identify entry points
 import logging
 logger = logging.getLogger(__name__)
 
-MEM_BAM_PERIPH_B    = 0xfff00000
-MSZ_BAM_PERIPH_B    = 0x00100000
-MEM_INTERNAL_FLASH  = 0x00000000
-MSZ_INTERNAL_FLASH  = 0x01000000
-MEM_EBI             = 0x20000000
-MSZ_EBI             = 0x01000000
-MEM_INTERNAL_SRAM   = 0x40000000
-MSZ_INTERNAL_SRAM   = 0x00040000
-MEM_PERIPH_BRIDGEA  = 0xC3F00000
-MSZ_PERIPH_BRIDGEA  = 0x00100000
+
+from vivisect.analysis.ppc.vlepages import DEFAULT_PPC_PAGES, DEFAULT_PPC_PAGES_VLE
+
+
+def set_rchw_vle(vw):
+    # If a VLE page configuration was specified manually, don't override it.
+    vle_pages = vw.getMeta('PpcVlePages')
+    if vle_pages is DEFAULT_PPC_PAGES
+        # If the VLE pages are still the default pages, update entries 1, 2,
+        # and 3 (Flash, EBI, SRAM) to be VLE.
+        vw.setMeta('PpcVlePages', DEFAULT_PPC_PAGES_VLE)
 
 
 def analyze(vw):
-    # TODO:  Boot Config 0, 1, 2, 3
-    if logger.getEffectiveLevel() < 40: vw.vprint('...analyzing PowerPC Entry Points.')
     logger.info('...analyzing PowerPC Entry Points.')
-    print('...analyzing PowerPC Entry Points.')
-
-    # add the PPC architecture structures to the namespace
-    vw.addStructureModule('ppc', 'vstruct.defs.ppc')
```

Comment:
where did we move this to?

Reply:
it was moved to the `archMarkupVW` function in the PPC architecture module. I don't remember why I moved the structure module though